packages/docker-engine/docker.service

```diff
@@ -10,6 +10,7 @@ StartLimitIntervalSec=60s
 
 [Service]
 Type=notify
+EnvironmentFile=/etc/network/proxy.env
```

Comment:
I just realized that if docker is now depending on settings, we need to restart it on changes, i.e. make a `docker` service in the model.

I guess the service has to be defined separately in the aws-dev and aws-ecs-1 defaults-overrides, since we probably shouldn't have it in the defaults..?

It should probably be restarted before ecs, in the ecs model, though, and we don't have dependencies between services.  Maybe in the ecs model, docker is just a restart-command in the ecs service, rather than being its own?

Reply:
> It should probably be restarted before ecs, in the ecs model, though, and we don't have dependencies between services.

There's maybe a race while you're configuring/restarting, but it shouldn't really matter?  The ECS agent should tolerate Docker going down and coming back up already.  There's also a dependency listed in the unit itself, though I know that our model isn't reading that.

Reply:
Ugh, this also applies to containerd, host-containerd, and kubelet, I believe.  We haven't dynamically affected the configuration of {host-,}containerd before.  kubelet we technically could, and it seems like it wouldn't have worked without a reboot, but they're mostly dynamic settings generated at boot that you wouldn't change.  (Maybe labels/taints are the worst offender?)

Reply:
@samuelkarp would containerd be similarly safe to restart?  No lost containers, logs, etc.?

Reply:
@bcressey found that restarting host-containerd kills SSH sessions, implying that the admin container is stopped.  Not sure if it's a necessary kill, or if our setup is wrong.

Reply:
> would containerd be similarly safe to restart? No lost containers, logs, etc.?

Containerd when used either directly through its API or with Docker does no networking and wouldn't need to be restarted.

Containerd when used with Kubernetes (through cri-containerd) does networking to pull images, and would need to be restarted.

By default, containerd will not stop containers when it exits, but systemd might be killing them.